### PR TITLE
Added an option to send slack message when a build passes

### PR DESCRIPTION
**Pull request by JennyChenyj**
Signed-off-by: Jenny Chen <Jenny.Chen@ibm.com>

### CODE REVIEW

**Comment by JennyChenyj** [comment]
Hey @AdamBrousseau! Thanks for reviewing it!

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy [hunk 1]
```diff
@@ -267,7 +267,10 @@ try {
                                     build(job_name, REPO, BRANCH, SHAS, OPENJ9_REPO, OPENJ9_BRANCH, OMR_REPO, OMR_BRANCH, SPEC, SDK_VERSION, BUILD_NODE, TEST_NODE, EXTRA_GETSOURCE_OPTIONS, EXTRA_CONFIGURE_OPTIONS, EXTRA_MAKE_OPTIONS, OPENJDK_CLONE_DIR, ADOPTOPENJDK_REPO, ADOPTOPENJDK_BRANCH, AUTOMATIC_GENERATION, CUSTOM_DESCRIPTION)
                                 }
                             }
-                        }
+                        }   
+                    }
+                    if (params.SLACK_ON_SUCCESS && SLACK_CHANNEL) {
+                        slackSend channel: SLACK_CHANNEL, color: 'good', message: "Build Passed: Started by ${JOB_NAME} #${BUILD_NUMBER} (<${BUILD_URL}|Open>)"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: I don't think we need the `Started by`, other than that it looks good to me.

Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: Also, this is in the wrong spot. It should be after the promote section. Where it is now, the message is sent right after the setup is complete before anything has run.

Comment (JennyChenyj) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: > Also, this is in the wrong spot. It should be after the promote section. Where it is now, the message is sent right after the setup is complete before anything has run.
Thanks for reviewing it! I will amend it. :)

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy [hunk 2]
```diff
@@ -267,7 +267,7 @@ try {
                                     build(job_name, REPO, BRANCH, SHAS, OPENJ9_REPO, OPENJ9_BRANCH, OMR_REPO, OMR_BRANCH, SPEC, SDK_VERSION, BUILD_NODE, TEST_NODE, EXTRA_GETSOURCE_OPTIONS, EXTRA_CONFIGURE_OPTIONS, EXTRA_MAKE_OPTIONS, OPENJDK_CLONE_DIR, ADOPTOPENJDK_REPO, ADOPTOPENJDK_BRANCH, AUTOMATIC_GENERATION, CUSTOM_DESCRIPTION)
                                 }
                             }
-                        }
+                        }
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: extra space here

**Comment by JennyChenyj** [comment]
Hey @AdamBrousseau! Thanks!

**Comment by pshipton** [comment]
jenkins compile plinux jdk8

**Pull request merged**